### ISSUE TITLE
Display Overlay Icons

### ISSUE THREAD

**Issue by oskretc** [fman-community/fman#354]
For some apps, its crucial to display overlay icons e.g. TortoiseSVN

**Comment by IgorKrupenja** [comment]
I think this was implemented already: I am seeing Dropbox overlay icons in fman 1.0.7 on Windows.

**Comment by mherrmann** [comment]
Do you happen to have TortoiseSVN installed as well @krupenja? If yes, do you see its overlays?

I did not work on this feature in recent releases. fman asks the operating system "give me the icon for this file". It appears that Windows gives fman the Dropbox-annotated icons. May not yet work for other apps / OSs.

**Comment by IgorKrupenja** [comment]
No, sorry, do not have TortoiseSVN so cannot test overlay icons for it - had no idea this could be app-specific.

**Comment by Ben-Kaniobi** [comment]
Sometimes OneDrive interferes with the TortoiseSVN overlays. Maybe this is the problem in your case @oskretc . As a workaround you could try this (at your own risk):

1. In the TortoiseSVN settings, locate _Icon Overlays > Overlay Handlers_ and click _Start registry editor_ (this should open in `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\ShellIconOverlayIdentifiers`.
2. Delete all `OneDrive` keys (folders) in `ShellIconOverlayIdentifiers` (e.g. `OneDrive1`, `OneDrive2`, etc.)
3. Log out and log in or reboot your PC

Note:
- I have not explicitly tested this with _fman_, but it fixed the problem in native windows explorer and FreeCommander when I had the problem in the past.
- I also don't use OneDrive, you will probably be missing the overlay icons there if you do this workaround and use OneDrive.
- OneDrive will prompt a message about completing the installation of synchronisation settings or something because it wants to restore them and needs the users permission to do that. You can discard this if you're happy with the workaround, or accept it to restore the registry settings.
- It might also be a good idea to first export `ShellIconOverlayIdentifiers` (right-click on it > export) so you could import it again (double click on the resulting .reg file) if it breaks something...

**Comment by oskretc** [comment]
This is partially working, I can see the overlayed icons for tortoise, the problem is that the icons never update, e.g. (when a change in the file is detected, tortoise will use a diff overlay) even across multiple fman restarts, the icons remain the same. I haven't pinpoint at what times the icons are initially fetched. (maybe they are cached at first launch).
